src/entity.ts

```diff
@@ -376,11 +437,45 @@ export namespace entity {
     return value instanceof entity.Key;
   }
 
+  /**
+   * Convert a protobuf `integerValue`.
+   *
+   * @private
+   * @param {object} value The `integerValue` to convert.
+   */
+  function decodeIntegerValue(value: ValueProto) {
+    const num = Number(value.integerValue);
+    if (!Number.isSafeInteger(num)) {
+      throw new Error(
+        'We attempted to return all of the numeric values, but ' +
+          (value.propertyName ? value.propertyName + ' ' : '') +
+          'value ' +
+          value.integerValue +
+          " is out of bounds of 'Number.MAX_SAFE_INTEGER'.\n" +
+          "Please consider passing 'options.wrapNumbers=true' and\n" +
```

Comment:
```diff
- and\n" +
+ or\n" +
```

?

Reply:
Oh, I see. I thought integerTypeCast options was its own thing. I’m not sure that I think we should require both wrapNumbers to be true as well as the full iTC object, since providing an iTC object should be enough to imply they want to handle conversion themselves.

Terminology wise, wrapping numbers is something that Datastore.Int does, but it’s not necessarily what the user function will do. I think they should be separate options the user has to process numbers, as opposed to being linked.

Reply:
The current theme is as follows:
1. Default not passing anything (not wrapping in DsInt) - API tries to convert to `Number`, throws on overflow.
    In case of an exception, user has to make another network call to retrieve the value.
2. Pass just `wrapNumbers=true` - API will return a `DsInt` obj in which
    - `#valueOf` will act the same way as default behavior, i.e. try to convert to `Number`, throw on overflow. However now, in case of an exception with `#valueOf` user can still retrieve a string representation of the value with `#value` and does not need to make another network call. That is described as "minimal number safety"
3. Along with `wrapNumbers=true` user also passing `iTCFunction` - API will return `DsInt` obj in which.
   - `#valueOf` API will use `iTCFunction` to convert the numeric values.

User's function is only referred to by `#valueOf` method which is DsInt component. That is why numbers need to be wrapped to be handled by `iTCFunction`. 

4. If user just sets `iTCFunction`, but not `wrapNumbers=true` - API will treat it as default behavior.

Reply:
We had a follow-up conversation about this in the chat— I had some suggestions, but I’m not sure you replied after that!